### PR TITLE
[PVR] Fix 'switch to previous channel' not working if it is in anothe…

### LINKED ISSUE #23928: [PVR] SwitchToPreviousChannel only works with channels from the same group
### Describe the bug

SwitchToPreviousChannel only works with channels from the same group.

## Expected Behavior

It should work with channels from different groups.

## Actual Behavior

If I play a channel, change the group with the channels OSD, play a channel from that group and press the '0' key to get back to the channel I played before, it doesn't work.

## Possible Fix

Modify the way CPVRChannelSwitchingInputHandler::SwitchToPreviousChannel() handles the channels switching.
Ideally, both the current group and channel should be changed. AFAIR this behavior has always been like this so I don't know if it is easily doable.

@ksooo Any thoughts?

### To Reproduce

Play a channel, change the group with the channels OSD, play a channel from that group and press the '0' key to get back to the channel that was played before: it doesn't work.

### Debuglog

[kodi.log](https://github.com/xbmc/xbmc/files/12905436/kodi.log)

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name:
   Windows NT x86 64-bit
 - Kodi version:
   Kodi (21.0-ALPHA3 (20.90.301) Git:20231012-94f0c210fe)

### PR DESCRIPTION
**Pull request by ksooo**
…r group than the currently playing channel.

Fixes #23928 

Fix: We must switch the active channel group before doing the actual channel switch in case new channel is not in the currently active group.

Runtime-tested on macOS, latest kodi master.

@phunkyfish could you have a look at the code change?

### CODE REVIEW

File: xbmc/pvr/guilib/PVRGUIActionsChannels.cpp [hunk 1]
```diff
@@ -77,6 +77,32 @@ void CPVRChannelSwitchingInputHandler::OnInputDone()
...
+{
+  const std::shared_ptr<CPVRPlaybackState> playbackState{
+      CServiceBroker::GetPVRManager().PlaybackState()};
+  const std::shared_ptr<CPVRChannelGroupsContainer> groups{
+      CServiceBroker::GetPVRManager().ChannelGroups()};
+  const std::shared_ptr<CPVRChannelGroup> group{
+      groups->Get(newChannel->IsRadio())->GetById(newChannel->GroupID())};
+
+  // Switch group if new channel is not in the active group.
+  if (group && group != playbackState->GetActiveChannelGroup(newChannel->IsRadio()))
```
Comment (enen92) [inline xbmc/pvr/guilib/PVRGUIActionsChannels.cpp:92]: Consider dropping the isRadio check and move it inside `GetActiveChannelGroup(const std::shared_ptr<CPVRChannelGroupMember>&)`. There's no need to give that info in advance, API consumers just want to know the active channel group regardless of the type.

**Review by enen92 — APPROVED** [review]

File: xbmc/pvr/guilib/PVRGUIActionsChannels.cpp (hunk 1 above)
Comment (enen92) [reply xbmc/pvr/guilib/PVRGUIActionsChannels.cpp:92]: Also `Active` is misleading, you're trying to get the channel group of the future channel to later switch to it, so definitely not active.

Comment (ksooo) [reply xbmc/pvr/guilib/PVRGUIActionsChannels.cpp:92]: I need to get the currently active group to be able to compare with the desired group. All good as it is.

Comment (ksooo) [reply xbmc/pvr/guilib/PVRGUIActionsChannels.cpp:92]: And to get the right type of group (radio or TV) I use the radio flag of the new channel. Welcome to PVR code base. :-)

Comment (enen92) [reply xbmc/pvr/guilib/PVRGUIActionsChannels.cpp:92]: > And to get the right type of group (radio or TV) I use the radio flag of the new channel. Welcome to PVR code base. :-)

Ok misread the line. Anyway, the bool argument could well go away :)

**Review by phunkyfish — APPROVED** [review]

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
Jenkins is acting up, but definitely nothing induced by this PR.

**Pull request merged**